src/sentry/models/sentryapp.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import absolute_import
+
+import six
+import uuid
+
+from bitfield import BitField
+from django.db import models
+from django.utils import timezone
+
+from sentry.db.models import (ArrayField, FlexibleForeignKey, ParanoidModel)
+from sentry.models import ApiScopes
+from sentry.utils.strings import dasherize
+
+
+class SentryApp(ParanoidModel):
+    __core__ = True
+
+    application = models.OneToOneField('sentry.ApiApplication',
+                                       related_name='sentry_app')
+
+    # Much of the OAuth system in place currently depends on a User existing.
+    # This "proxy user" represents the SentryApp in those cases.
+    proxy_user = models.OneToOneField('sentry.User',
+                                      related_name='sentry_app')
+
+    # The owner is an actual Sentry User who created the SentryApp. Used to
+    # determine who can manage the SentryApp itself.
+    owner = FlexibleForeignKey('sentry.User',
+                               related_name='owned_sentry_apps')
+
+    # The set of OAuth scopes necessary for this integration to function.
+    scopes = BitField(flags=ApiScopes().to_bitfield())
+    scope_list = ArrayField(of=models.TextField())
+
+    name = models.TextField()
+    slug = models.TextField()
```

Comment:
This doesn't need to be unique, right?

Reply:
Probably should, now that I think about it.

Reply:
If this is going to be indexed, you'll need to change it to CharField from TextField. This is a limitation of at least MySQL, in that you can't index a full TextField. Postgres can, but MySQL cannot. But a length of like, 64 should be plenty sufficient?